homeassistant/components/plex/server.py

```diff
@@ -78,13 +88,92 @@ def _connect_with_url():
 
         _connect_with_url()
 
-    def clients(self):
-        """Pass through clients call to plexapi."""
-        return self._plex_server.clients()
+    def update_platforms(self):
+        """Update the platform entities."""
+        available_client_ids = []
+        new_plex_clients = []
+
+        try:
+            devices = self._plex_server.clients()
+        except plexapi.exceptions.BadRequest:
+            _LOGGER.exception("Error listing plex devices")
+            return
+        except requests.exceptions.RequestException as ex:
+            _LOGGER.warning(
+                "Could not connect to Plex server: %s (%s)", self.friendly_name, ex
+            )
+            return
+
+        for device in devices:
+            available_client_ids.append(device.machineIdentifier)
+
+            if device.machineIdentifier not in self._clients:
+                new_client = PlexMediaPlayer(
+                    self, device, None, self._sessions, self.update_platforms
+                )
+                self._clients[device.machineIdentifier] = new_client
+                _LOGGER.debug("New device: %s", device.machineIdentifier)
+                new_plex_clients.append(new_client)
```

Comment:
Entities should not be stored in a shared container. When using the dispatch helper it's usually only needed to store a reference id to the entity, ie the entity_id.